packages/core/src/tabs/tab-group.ts

```diff
@@ -0,0 +1,248 @@
+import type { CSSResultGroup, TemplateResult } from 'lit';
+import type { EventEmitter } from '../utils/decorators/event.js';
+import type { ScopedElementsMap } from '@open-wc/scoped-elements';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements';
+import { LitElement, html } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { event } from '../utils/decorators/event.js';
+import { RovingTabindexController } from '../utils/controllers/roving-tabindex.js';
+import { Tab } from './tab.js';
+import styles from './tab-group.scss.js';
+import { TabPanel } from './tab-panel.js';
+
+let tabGroupCount = 0;
+
+export class TabGroup extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-tab': Tab,
+      'sl-tab-panel': TabPanel
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /**
+   * Unique ID for each tab group component present.
+   */
+  private tabGroupId = `sl-tab-group-${tabGroupCount++}`;
+
+  private observer?: MutationObserver;
+
+  private static observerOptions = {
+    attributes: true,
+    subtree: true,
+    attributeFilter: ['selected'],
+    attributeOldValue: true
+  };
+
+  /** The slotted tabs. */
+  @queryAssignedElements({ slot: 'tabs' }) tabs?: Tab[];
+
+  #rovingTabindexController = new RovingTabindexController<Tab>(this, {
+    focusInIndex: (elements: Tab[]) => elements.findIndex(el => el.selected),
+    elements: () => this.tabs || [],
+    isFocusableElement: (el: Tab) => !el.disabled
+  });
+
+  /**
+   * The current tab node selected in the tab group.
+   */
+  @state() private selectedTab: Tab | null = this.initialSelectedTab;
+
+  @event() tabChange!: EventEmitter<number>;
+
+  @property({ reflect: true }) vertical = false;
```

Comment:
Please add a comment for each `@property`; that is then used for documentation later on.